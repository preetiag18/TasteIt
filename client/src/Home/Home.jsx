import React from "react";
import { NavLink } from "react-router-dom";

import Hero from "../Hero/Hero";

import classes from "./Home.module.css";
const Home = () => {
  return (
    <div>
      <Hero />
      <div className={classes.moreDetailsContainer}>
        <div className={classes.moreDetails}>
          <h3>Browse recipe</h3>
          <p>Find your favorite in this collection. You can seatmm√∂,......</p>
          <NavLink to="recipes">All recipes</NavLink>
        </div>
        <div className={classes.moreDetails}>
          <h3>Add recipe</h3>
          <p>
            Recipe from your country misssing...!! , no worries, you can add
            them now
          </p>
          <NavLink to="addnewrecipes">Add new recipes</NavLink>
        </div>
        <div className={classes.moreDetails}>
          <h3>Want to know more about our projects</h3>
          <p>Visit our website</p>
          <a href="https://bc.fi" target="_blank" rel="noreferrer">
            Business college helsinki homepage
          </a>
        </div>
      </div>
    </div>
  );
};

export default Home;
