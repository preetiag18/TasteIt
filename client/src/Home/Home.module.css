.moreDetailsContainer {
  margin: 4rem;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  gap: 1rem;
}

.moreDetails {
  border: 1px solid black;
  max-width: 20rem;
  padding: 3rem;
}

.moreDetails > * {
  margin-bottom: 1rem;
}

a {
  text-decoration: none;
}
