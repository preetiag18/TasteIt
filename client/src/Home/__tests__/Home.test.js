import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import Home from "../Home";
import { BrowserRouter } from "react-router-dom";

describe("Home", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
