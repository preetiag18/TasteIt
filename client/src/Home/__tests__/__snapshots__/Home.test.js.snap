// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div>
  <div
    className="container"
  >
    <video
      autoPlay={true}
      className="bgVideo"
      loop={true}
      muted={true}
    >
      <source
        src="Rice.mp4"
        type="video/mp4"
      />
    </video>
    <div
      className="caption"
    >
      <h1
        className="heading large_font"
      >
        TasteIt
      </h1>
      <h2
        className="heading medium_font"
      >
        TasteIt is the recipe app which is made in React22S group lessons
      </h2>
      <a
        className="heading"
        href="/recipes"
        onClick={[Function]}
      >
        <button
          className="button"
        >
          Browse recipes
        </button>
      </a>
    </div>
  </div>
  <div
    className="moreDetailsContainer"
  >
    <div
      className="moreDetails"
    >
      <h3>
        Browse recipe
      </h3>
      <p>
        Find your favorite in this collection. You can seatmm√∂,......
      </p>
      <a
        className=""
        href="/recipes"
        onClick={[Function]}
      >
        All recipes
      </a>
    </div>
    <div
      className="moreDetails"
    >
      <h3>
        Add recipe
      </h3>
      <p>
        Recipe from your country misssing...!! , no worries, you can add them now
      </p>
      <a
        className=""
        href="/addnewrecipes"
        onClick={[Function]}
      >
        Add new recipes
      </a>
    </div>
    <div
      className="moreDetails"
    >
      <h3>
        Want to know more about our projects
      </h3>
      <p>
        Visit our website
      </p>
      <a
        href="https://bc.fi"
        rel="noreferrer"
        target="_blank"
      >
        Business college helsinki homepage
      </a>
    </div>
  </div>
</div>
`;
